YesGraph/yesgraph-superwidget
# Update deploy process

### What’s this PR do?
This PR makes a few updates to the deploy process, with the goal of allowing more thorough testing before each release.

Specifically, it allows us to deploy separately to 3 locations:
 1. the [CDN root](https://cdn.yesgraph.com/yesgraph-invites.js) (for production)
 2. the [ "dev/" branch of the CDN](https://cdn.yesgraph.com/dev/yesgraph-invites.js) (for testing changes live, before updating the root CDN), and
 3. to a local "deployed/" folder (for testing the deploy process itself, without actually updating the CDN).

### Where should the reviewer start?
Start by reviewing the updated deploy task in gulp/tasks/deploy.js

### How should this be manually tested?
Try running a test deploy! Use the `--test` flag to indicate that instead of pushing to S3 we should just save the updated files locally, and the `--prod` flag to indicate that we should process all files, (not just the dev/ files).

All in all the command is this:
`gulp deploy --test --prod`

You should now have a whole bunch of files in the new deployed/ folder:
```
deployed/
├── dev
│   ├── v0.0.6
│   │   ├── yesgraph-invites.css
│   │   └── yesgraph-invites.min.css
│   ├── v0.1.7
│   │   ├── yesgraph.js
│   │   ├── yesgraph.min.js
│   │   └── yesgraph.min.js.map
│   ├── v1.1.6
│   │   ├── yesgraph-invites.js
│   │   ├── yesgraph-invites.min.js
│   │   └── yesgraph-invites.min.js.map
│   ├── yesgraph-invites.css
│   ├── yesgraph-invites.js
│   ├── yesgraph-invites.min.css
│   ├── yesgraph-invites.min.js
│   ├── yesgraph-invites.min.js.map
│   ├── yesgraph.js
│   ├── yesgraph.min.js
│   └── yesgraph.min.js.map
├── v0.0.6
│   ├── yesgraph-invites.css
│   └── yesgraph-invites.min.css
├── v0.1.7
│   ├── yesgraph.js
│   ├── yesgraph.min.js
│   └── yesgraph.min.js.map
├── v1.1.6
│   ├── yesgraph-invites.js
│   ├── yesgraph-invites.min.js
│   └── yesgraph-invites.min.js.map
├── yesgraph-invites.css
├── yesgraph-invites.js
├── yesgraph-invites.min.css
├── yesgraph-invites.min.js
├── yesgraph-invites.min.js.map
├── yesgraph.js
├── yesgraph.min.js
└── yesgraph.min.js.map

7 directories, 32 files
```
### What are the relevant tickets?
[Update Superwidget deploy process](https://app.asana.com/0/59202558034519/162364139954367)

### Does the knowledge base need an update?
Yeah, I should add some documentation to the wiki.
